src/GitTfs/Core/TfsInterop/TfsPlugin.cs

```diff
@@ -23,7 +23,8 @@ public static TfsPlugin Find()
                 return pluginLoader.TryLoadVsPluginVersion(explicitVersion) ??
                        pluginLoader.Fail("Unable to load TFS version specified in GIT_TFS_CLIENT (" + explicitVersion + ")!");
             }
-            return pluginLoader.TryLoadVsPluginVersion("2015", true) ??
+            return pluginLoader.TryLoadVsPluginVersion("2017", true) ??
+                   pluginLoader.TryLoadVsPluginVersion("2017") ??
                    pluginLoader.TryLoadVsPluginVersion("2015") ??
```

Comment:
Hm. I am not sure this logic works for VS2017 or VS2019, as the check for "is corresponding VS version installed" checks
for the existence of the registry entries
```
HKLM\SOFTWARE\Wow6432Node\Microsoft\DevDiv\vs\Servicing\<versionCode>
HKLM\SOFTWARE\Microsoft\DevDiv\vs\Servicing\<versionCode>
```
where `versionCode` would be 15.0 for VS2017.
But for e.g. VS2019 both keys would not be found with versionCode=16.0, because to the best of my knowledge, there is nothing there which could indicate that VS2019 is even installed.
For VS2017, I have the key, but I am not sure if this is by accident only. Could someone pls double check if either one of those keys exist for a VS2017 installation?


Currently, I am leaning to remove the optional isVisualStudioRequired parameter for `public TfsPlugin TryLoadVsPluginVersion(string version, bool isVisualStudioRequired = false)` altoegether, especially as it will not work for VS2019 anyhow.

Reply:
I think removing `isVisualStudioRequired` would make sense.

The intent of the current code is to try to automatically figure out which version of VS you're actually using on your computer, so that we are running in an env where you have checkin policies configured. (In the ancient past, TFVC assemblies weren't distributed with git-tfs, so git-tfs needed to probe around to see which version it was possible to load at all.)

All of this would probably make sense to replace with configuration. IOW, git-tfs could default to using the newest version unless you override it with `git config --local git-tfs.client-version VS2015` or `git config --global git-tfs.client-version VS2015`. I'd be :+1: on removing support for the `GIT_TFS_CLIENT` env var.

How does that sound?

Reply:
I already started working on removing `isVisualStudioRequired` yesterday, anticipating your answer 😄 

But I have currently no intention to implement the config based approach of selecting the version, at least not in this PR,
as it is orthogonal to the PRs goal and we already have the established and documented way to select a different VS version with the `GIT_TFS_CLIENT` environment variable. 
Don't get me wrong, I **like** what you are proposing, but I don't want to overextend the scope of this PR

Reply:
> But I have currently no intention to implement the config based approach of selecting the version, at least not in this PR,
> as it is orthogonal to the PRs goal and we already have the established and documented way to select a different VS version with the `GIT_TFS_CLIENT` environment variable.
> Don't get me wrong, I **like** what you are proposing, but I don't want to overextend the scope of this PR

:+1: :+1: